.github/workflows/pr-merged.yml

```diff
@@ -1,11 +1,9 @@
-name: Create Dev Docker Images
+name: Pull Request Merged to Main
 
 on:
-  schedule:
-    # Run cron job at 8AM Monday to Sunday.
-    - cron: '0 8 * * *'
-  workflow_dispatch:
-  pull_request:
+  push:
+    branches:
+      - main
```

Comment:
Also means there wont be latest docker images available in AWS for the rasa-private workflow [here](https://github.com/RasaHQ/rasa-private/blob/main/.github/workflows/continous-integration.yml#L253) . We don't work on sdk code too much so this push to main will rarely happen.

Reply:
What is on `main` is what should be in the `latest` docker image built. 
I added a job which will build Docker image and tag it with pr number.
The AWS rule will not delete this image because it is tagged.
Devs which are working on `rasa-private` repo can use PR tagged Docker image (which will not be deleted) unless PR is closed. Once a PR on Rasa SDK is merged dev can switch to `latest` Docker image

Previous workflow would have race-condition which would overwrite `latest` image in case there were two or more PRs opened on the SDK.